bundles/org.openhab.binding.millheat/src/main/java/org/openhab/binding/millheat/internal/handler/MillheatHeaterHandler.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.millheat.internal.handler;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.millheat.internal.MillheatBindingConstants;
+import org.openhab.binding.millheat.internal.config.MillheatHeaterConfiguration;
+import org.openhab.binding.millheat.internal.model.Heater;
+import org.openhab.binding.millheat.internal.model.MillheatModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import tec.uom.se.unit.Units;
+
+/**
+ * The {@link MillheatHeaterHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Arne Seime - Initial contribution
+ */
+@NonNullByDefault
+public class MillheatHeaterHandler extends MillheatBaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(MillheatHeaterHandler.class);
+    private @NonNullByDefault({}) MillheatHeaterConfiguration config;
+    private @NonNullByDefault({}) MillheatAccountHandler accountHandler;
+
+    public MillheatHeaterHandler(Thing thing) {
+        super(thing);
+        accountHandler = (MillheatAccountHandler) getBridge();
```

Comment:
please don't do that. There is no guarantee that the bridge handler is not changing

Reply:
Is it necessary to check that a bridge is available? By config the Thing cannot exist without one?

Reply:
That is true. But you can define the thing and then delete the bridge. The thing will go OFFLINE of course because the bridge is missing. Then you re-add a Bridge with the same UID. The bridge handler will be different but your thing handler will not notice that because it cached the original one.